scripts/scil_compute_bundle_profiles.py

```diff
@@ -0,0 +1,131 @@
+
+
+def average_euclidean(x, y):
+    return np.mean(norm_l2(x - y))
+
+
+def _build_arg_parser():
+    p = argparse.ArgumentParser(formatter_class=argparse.RawTextHelpFormatter,
+                                description=__doc__)
+    p.add_argument('in_bundle',
+                   help='Fiber bundle file to compute the bundle profiles on.')
+    p.add_argument('in_metrics', nargs='+',
+                   help='Nifti metric(s) on which to compute '
+                        'the bundle profiles.')
+
+    g = p.add_mutually_exclusive_group()
+    g.add_argument('--in_centroid',
+                   help='If provided it will be used to make sure all '
+                        'streamlines go in the same direction. \nOtherwise, a '
+                        'centroid will be automatically computed.\n'
+                        'Number of point per streamline will be set according'
+                        ' to centroid.')
+    g.add_argument('--nb_pts_per_streamline',
+                   type=int, default=20,
+                   help='Subsample each streamline to this number of points '
+                        '[%(default)s].')
+
+    add_json_args(p)
+    add_reference_arg(p)
+    add_overwrite_arg(p)
+    return p
+
+
+def main():
+    parser = _build_arg_parser()
+    args = parser.parse_args()
+
+    assert_inputs_exist(parser, [args.in_bundle] + args.in_metrics,
+                        optional=args.in_centroid)
+
+    if args.nb_pts_per_streamline <= 1:
+        parser.error('--nb_pts_per_streamline {} needs to be greater than '
+                     '1'.format(args.nb_pts_per_streamline))
+
+    assert_same_resolution(args.in_metrics)
+    metrics = [nib.load(m) for m in args.in_metrics]
+
+    sft = load_tractogram_with_reference(parser, args, args.in_bundle)
+
+    bundle_name, _ = os.path.splitext(os.path.basename(args.in_bundle))
+    stats = {}
+    if len(sft) == 0:
+        stats[bundle_name] = None
+        print(json.dumps(stats, indent=args.indent, sort_keys=args.sort_keys))
+        return
+
+    # Centroid - will be use as reference to reorient each streamline
+    if args.in_centroid:
+        sft_centroid = load_tractogram_with_reference(parser, args,
+                                                      args.in_centroid)
+        centroid_streamlines = sft_centroid.streamlines[0]
+        nb_pts_per_streamline = len(centroid_streamlines)
+    else:
+        centroid_streamlines = get_streamlines_centroid(sft.streamlines,
+                                                        args.nb_pts_per_streamline)
+        nb_pts_per_streamline = args.nb_pts_per_streamline
+
+    resampled_sft = resample_streamlines_num_points(sft, nb_pts_per_streamline)
+
+    # Make sure all streamlines go in the same direction. We want to make
+    # sure point #1 / args.nb_pts_per_streamline of streamline A is matched
+    # with point #1 / 20 of streamline B and so on
+    num_streamlines = len(resampled_sft.streamlines)
```

Comment:
Related to previous comment `len(sft)` and `len(sft.streamlines)` are the same, but one is a bit clearer